nucleus/deployment/common/src/main/java/com/sun/enterprise/deploy/shared/FileArchive.java

```diff
@@ -799,15 +823,15 @@ public static void markDeletedArchive(final File archiveFile) {
                     return;
                 }
 
-                final URI archiveURI = archiveFile.toURI();
+                final Path archivePath = archiveFile.toPath();
                 PrintStream ps = null;
                 try {
                     ps = new PrintStream(markerFile(archiveFile));
                 } catch (FileNotFoundException ex) {
                     throw new RuntimeException(ex);
                 }
                 for ( ; staleFileIt.hasNext(); ) {
```

Comment:
How about moving this for loop into the above try block and use a try-with-resources to close the PrintStream automatically?